### PR TITLE
refactor release mechanism to release whole project as one

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Refactoring
- Documentation

### Description

Following discussion on #672 we decided that kroxylicious project will be released as one (no separate API module release).
This change unifies the releases scripts into one and eliminates the idea of a separate API version number.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+cleanup() {
+    if [[ -n ${ORIGINAL_WORKING_BRANCH} ]]; then
+        git checkout "${ORIGINAL_WORKING_BRANCH}" || true
+    fi
+
+    if [[ ${TEMPORARY_RELEASE_BRANCH} ]]; then
+        # Note that git branch -D echos the sha of the deleted branch to
+        # stdout.  This is great for debugging the release process as it
+        # lets the developer restore to the state of the tree.
+        git branch -D "${TEMPORARY_RELEASE_BRANCH}" || true
```
Comment (k-wall) [inline scripts/release.sh:58]: You'll see:
```
Deleted branch prepare-release-2023-10-22 (was 1af9612b).
Deleted tag 'v0.3.0' (was 1af9612b)
```

so you can `git checkout 1af9612b` to get back to the state created by the release script.

File: RELEASING.md [hunk 2]
```diff
@@ -26,20 +26,21 @@ TIP: To get the short ID for your key `gpg2 --show-keys --keyid-format short ${Y
...
 ```shell
-scripts/release-api.sh ${RELEASE_VERSION}
+./scripts/release.sh -k <YOUR_KEY_SHORT_ID> -v <RELEASE_VERSION> [-b <BRANCH_FROM>] [-r <REPOSITORY>]
 ```
 
-```shell
-scripts/release-framework.sh ${RELEASE_VERSION}
-```
+where
+* `<YOUR_KEY_SHORT_ID>` is the short id of your PGP key
```
Comment (franvila) [inline RELEASING.md:36]: I think you mean "GPG" key

**Review by franvila — APPROVED** [review]
Once we have a release script for everything, is it intended to create a jenkins pipeline to run the releases instead of doing it manually?

**Comment by k-wall** [comment]
> Once we have a release script for everything, is it intended to create a jenkins pipeline to run the releases instead of doing it manually?

Yes, certainly.  For this project and the extension.  I hope we can do this soon.

File: RELEASING.md (hunk 2 above)
Comment (k-wall) [reply RELEASING.md:36]: PGP is the *requirement*,  but users are actually free to use pgp, gpg or any other pgp compatible.  We talk about PGP higher up in the doc, so this new text is consistent.   

https://www.gnupg.org/
> GnuPG is a complete and free implementation of the OpenPGP standard as defined by [RFC488](https://www.ietf.org/rfc/rfc4880.txt)






OpenPGP

**Pull request merged**